### PR TITLE
Added documentation about pooled events

### PR DESCRIPTION
**Pull request by edvinerikson**
Added documentation about pooled events, some feedback is very welcome. 
Related to #4514 
cc @jimfb

### CODE REVIEW

File: docs/docs/ref-05-events.md [hunk 1]
```diff
@@ -33,11 +33,36 @@ string type
...
+  }, 0);
+
+  this.setState({clickEvent: event}); // Won't work. this.state.clickEvent will only contain null values.
+  this.setState({eventType: event.type}); // You can still export event properties.
+}
+```
+
+> Note:
+>
+> If you want to access event properties in an asynchronous way, you need to create a new object e.g `var copiedEvent = assign({}, event);` or assign the specific properties to variables.
```
Comment (jimfb) [inline docs/docs/ref-05-events.md]: If you want to access the event properties in an asynchronous way, you should call `event.persist()` on the event, which will remove the synthetic event from the pool and allow references to the event to be retained by user code.

**Comment by jimfb** [comment]
Looks pretty good.  One minor thing: we should mention `event.persist()` instead of having the user manually copy the event.

I'm also going to let the PR sit for a couple days so everyone has a chance to take a look before we merge.

**Comment by edvinerikson** [comment]
Thanks for the feedback.
Sounds good to me. :+1:

**Comment by jimfb** [comment]
@edvinerikson Thanks!

**Pull request merged**